### PR TITLE
MGMT-7593: Added custom manifest feature usage

### PR DESCRIPTION
**Pull request by sagidayan**
# Assisted Pull Request

## Description

In order to know that a user has requested to add a custom manifest, for metrics counting for user usage (and not internal usage), We needed to change all internal manifests creations calls to use the internal function directly instead of calling the API handler.

Currently we only add a usage on create and we don't remove it on delete. This is because we do need to discuss this flow. If a user override a built in manifest and deletes it etc.

The manifests manager has the ability to remove a feature usage with the new `setUsage` function - just not yet called from the delete API.

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @slaviered 
/cc @masayag 
/cc @osherdp 

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/manifests/manifests.go [hunk 1]
```diff
@@ -54,13 +49,18 @@ type Manifests struct {
...
 	objectHandler s3wrapper.API
+	usageAPI      usage.API
 }
 
 func (m *Manifests) CreateClusterManifest(ctx context.Context, params operations.CreateClusterManifestParams) middleware.Responder {
 	manifest, err := m.CreateClusterManifestInternal(ctx, params)
 	if err != nil {
 		return common.GenerateErrorResponder(err)
 	}
+	err = m.setUsage(true, manifest, params.ClusterID)
```
Comment (masayag) [inline internal/manifests/manifests.go:60]: a similar call with active=false should be added to `DeleteClusterManifest`

Comment (sagidayan) [reply internal/manifests/manifests.go:60]: @masayag As mentioned in the PR, I added this for easy-later-on-implementation of the delete manifest. But for now we only add usage not removing.

But this is a good point - and we need to discuss how to handle manifest deletions.

File: internal/manifests/api/interface.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/restapi"
+	operations "github.com/openshift/assisted-service/restapi/operations/manifests"
+)
+
+//go:generate mockgen -package api -destination mock_manifests_api.go . ManifestsAPI
+type ManifestsAPI interface {
+	restapi.ManifestsAPI
+	ClusterManifestsInternals
```
Comment (ybettan) [inline internal/manifests/api/interface.go:14]: Why don't we just use `CreateClusterManifestInternal`, `ListClusterManifestsInternal` and `DeleteClusterManifestInternal` directly here?

File: internal/network/manifests_generator_test.go [hunk 3]
```diff
@@ -548,15 +566,37 @@ var _ = Describe("disk encryption manifest", func() {
...
 			Expect(db.Create(&c).Error).NotTo(HaveOccurred())
-
-			mockManifestsApi.EXPECT().CreateClusterManifest(ctx, gomock.Any()).Return(operations.NewCreateClusterManifestCreated()).Times(t.numOfManifests)
-			err := manifestsGeneratorApi.AddDiskEncryptionManifest(ctx, log, &c)
 			if t.shouldErr == false {
+				if t.diskEncryption != nil {
+					switch t.diskEncryption.EnableOn {
+					case models.DiskEncryptionEnableOnAll:
+						mockManifestsApi.EXPECT().CreateClusterManifestInternal(ctx, gomock.Any()).Return(&models.Manifest{
+							FileName: generateManifestFileName("worker", t.diskEncryption.Mode),
```
Comment (ybettan) [inline internal/network/manifests_generator_test.go:576]: Did you mean `master` ?

Comment (sagidayan) [reply internal/network/manifests_generator_test.go:576]: yes, thanks 👏🏽

**Review by osherdp — APPROVED** [review]

**Review by osherdp — APPROVED** [review]

**Review by osherdp — APPROVED** [review]

**Review by osherdp — APPROVED** [review]

**Comment by osherdp** [comment]
```
curl: (6) Could not resolve host: releases-art-rhcos.svc.ci.openshift.org
```
looks like a temporal DNS failure. Let's try to retest
/test e2e-metal-assisted-operator-ztp

**Pull request merged**